### PR TITLE
Update taglibs for JSTL tests

### PR DESCRIPTION
**Pull request by alwin-joseph**
**Fixes Issue**
None specified

**Related Issue(s)**
https://github.com/eclipse-ee4j/jstl-api/issues/144

**Describe the change**
Updated the below taglibs. This will fix around 400 failing JSTL standalone TCK tests with glassfish 7
```
http://java.sun.com/jsp/jstl/sql -> jakarta.tags.sql
http://java.sun.com/jsp/jstl/functions -> jakarta.tags.functions
http://java.sun.com/jsp/jstl/xml -> jakarta.tags.xml

http://java.sun.com/jsp/jstl/core -> jakarta.tags.core
http://java.sun.com/jsp/jstl/fmt -> jakarta.tags.fmt
```

### CODE REVIEW

**Comment by BalusC** [comment]
I'm surprised to see this. They are at least backwards compatible in XHTML files. I can't tell for JSP files.

Do all tests pass at your environment now?

**Comment by alwin-joseph** [comment]
> 

> I'm surprised to see this. They are at least backwards compatible in XHTML files. I can't tell for JSP files.
> 

The logs with failures  https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/master/1667/artifact/jstltck.log had exceptions  like 
`org.glassfish.wasp.WaspException: /positiveTransformXsltSystemIdTest.jsp&#40;19,62&#41; PWC6188: The absolute uri: http://java.sun.com/jsp/jstl/core cannot be resolved in either web.xml or the jar files deployed with this application`

hence I chose to update them.

> Do all tests pass at your environment now?

Before this change :

[javatest.batch] ********************************************************************************
[javatest.batch] Completed running 542 tests.
[javatest.batch] Number of Tests Passed      = 87
[javatest.batch] Number of Tests Failed      = 455
[javatest.batch] Number of Tests with Errors = 0
[javatest.batch] ********************************************************************************

After this change : 

[javatest.batch] ********************************************************************************
[javatest.batch] Completed running 542 tests.
[javatest.batch] Number of Tests Passed      = 468
[javatest.batch] Number of Tests Failed      = 74
[javatest.batch] Number of Tests with Errors = 0
[javatest.batch] ********************************************************************************

latest run logs : https://ci.eclipse.org/jakartaee-tck/blue/rest/organizations/jenkins/pipelines/jakartaee-tck-alw/branches/jstl-taglibupdate/runs/3/nodes/51/steps/56/log/?start=0 

Now most of the failures are with exceptions like  

`org.glassfish.wasp.WaspException: jakarta.el.PropertyNotFoundException: ELResolver cannot handle a null base Object with identifier &#39;str&#39;</pre></p><p><b>root cause</b> <pre>jakarta.el.PropertyNotFoundException: ELResolver cannot handle a null base Object with identifier`

and 

`The server encountered an internal error that prevented it from fulfilling this request.</p><p><b>exception</b> <pre>jakarta.servlet.ServletException: In &amp;lt&#59;parseDate&amp;gt&#59;, value attribute can not be parsed: &quot;Nov 21, 2000, 3:45:02 AM&quot;</pre></p><p><b>root cause</b> <pre>java.text.ParseException: Unparseable date: &quot;Nov 21, 2000, 3:45:02 AM&quot`

**Comment by BalusC** [comment]
> exceptions like `org.glassfish.wasp.WaspException`

Ah okay, those are indeed exclusively coming from JSP files.

**Review by BalusC — APPROVED** [review]

File: src/web/jsf/spec/jstl/fntrim/fntrim.xhtml [hunk 1]
```diff
@@ -22,7 +22,7 @@
       xmlns:h="http://java.sun.com/jsf/html"
       xmlns:f="http://java.sun.com/jsf/core"
       xmlns:c="http://java.sun.com/jsp/jstl/core"
```
Comment (BalusC) [inline src/web/jsf/spec/jstl/fntrim/fntrim.xhtml:24]: Not a blocker (test will pass just fine as Jakarta Faces has currently still backwards compatibility for old XML namespaces), but for the record all those java.sun.com XML namespaces here above should preferably be migrated to the new ones as well. The backwards compatibility may be removed in a future Jakarta Faces version.

File: src/web/jsf/spec/jstl/foreachtag/foreachtag_facelet.xhtml [hunk 2]
```diff
@@ -23,7 +23,7 @@
       xmlns:h="http://java.sun.com/jsf/html"
       xmlns:f="http://java.sun.com/jsf/core"
       xmlns:c="http://java.sun.com/jsp/jstl/core"
```
Comment (BalusC) [inline src/web/jsf/spec/jstl/foreachtag/foreachtag_facelet.xhtml:25]: Also here.

File: src/web/jsf/spec/jstl/fntrim/fntrim.xhtml (hunk 1 above)
Comment (alwin-joseph) [reply src/web/jsf/spec/jstl/fntrim/fntrim.xhtml:24]: These are actually jsf test files, I will create a separate PR  to update taglibs for all jsf files remaining. Thanks.

File: src/web/jsf/spec/jstl/foreachtag/foreachtag_facelet.xhtml (hunk 2 above)
Comment (alwin-joseph) [reply src/web/jsf/spec/jstl/foreachtag/foreachtag_facelet.xhtml:25]: will address in a separate PR.

**Comment by alwin-joseph** [comment]
@BalusC Thanks for the review.

**Pull request merged**